### PR TITLE
Merge Debug functionality into master

### PR DESCRIPTION
**Pull request by orilevari**
This pull request sets up the ability to specify nodes in the edit view to capture 'debug' information either in png or text format. Then when you run the model in the run tab, you can choose to capture "debug" and then the graph will be behind the hood modified to contain debug operators and ran against a custom Winml provider with the debug operator registered. Then the output data can be saved to disk as specified by the user after a successful run.

### CODE REVIEW

File: Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        std::string line;
+        if (!std::getline(fileStream, line))
+        {
+            ThrowFailure(L"BindingUtilities: expected more input rows.");
+        }
+
+        // Split the line into strings for each value.
+        std::istringstream elementsString(line);
+        std::string elementString;
+        while (std::getline(elementsString, elementString, ','))
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h]: Make sure there is no space in the line.

File: Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        return elementStrings;
+    }
+
+    // Binds tensor floats, ints, doubles from CSV data.
+    ITensor CreateBindableTensor(const ILearningModelFeatureDescriptor& description, std::wstring inputPath)
+    {
+        auto name = description.Name();
+        auto tensorDescriptor = description.try_as<TensorFeatureDescriptor>();
+
+        if (!tensorDescriptor)
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h]: Tensor is also bindable to imageFeatureDescriptor.

File: Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+        if (!imageDescriptor)
+        {
+            std::cout << "BindingUtilities: Input Descriptor type isn't tensor." << std::endl;
+            throw;
+        }
+
+        auto softwareBitmap = LoadImageFile(imageDescriptor, inputDataType, imagePath.c_str());
+
+		auto videoFrame = VideoFrame::CreateWithSoftwareBitmap(softwareBitmap);
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/BindingUtilities.h]: remove extra spaces.

File: Tools/WinMLDashboard/src/cpp/DebugRunner/DebugRunner.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	static bool Run(std::wstring modelPath, std::wstring inputPath, LearningModelDeviceKind kind)
+	{
+		std::wcout << modelPath;
+		auto customOperatorProvider = winrt::make<DebugOperatorProvider>();
+		auto provider = customOperatorProvider.as<ILearningModelOperatorProvider>();
+		auto model = LearningModel::LoadFromFilePath(modelPath, provider);
+		LearningModelSession session(model, LearningModelDevice(kind));
+		LearningModelBinding binding(session);
+
+		auto&& description = model.InputFeatures().GetAt(0);
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/DebugRunner.cpp]: What if there are more than one inputs in the model.

Comment (zhangxiang1993) [reply Tools/WinMLDashboard/src/cpp/DebugRunner/DebugRunner.cpp]: I think the input Path box in Run view need to handle multi-input case

File: Tools/WinMLDashboard/src/cpp/DebugRunner/ModelBinding.h [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            ThrowFailure(L"ModelBinding: Binding feature type not implemented");
+        }
+    }
+
+    winrt::Windows::AI::MachineLearning::ILearningModelFeatureDescriptor GetDesc()
+    {
+        return m_bindingDesc;
+    }
+
+    UINT GetNumElements() const
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/ModelBinding.h]: use uint32_t.

File: Tools/WinMLDashboard/src/cpp/DebugRunner/ModelBinding.h [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        return m_numElements;
+    }
+
+    UINT GetElementSize() const
+    {
+        return m_elementSize;
+    }
+
+    std::vector<int64_t> GetShapeBuffer()
```
Comment (zhangxiang1993) [inline Tools/WinMLDashboard/src/cpp/DebugRunner/ModelBinding.h]: replace int64_t, UINT to uint32_t.

File: Tools/WinMLDashboard/src/cpp/DebugRunner/DebugRunner.cpp (hunk 4 above)
Comment (orilevari) [reply Tools/WinMLDashboard/src/cpp/DebugRunner/DebugRunner.cpp]: I think that handling multiple inputs would be good as well. I migrated most of the logic of the DebugRunner code over from WinMLRunner so perhaps we should have this discussion in regards to WinMLRunner functionality as well.

**Pull request merged**